packages/pigeon/lib/cpp_generator.dart

```diff
@@ -437,6 +413,89 @@ else if (const int64_t* ${pointerFieldName}_64 = std::get_if<int64_t>(&$encodabl
   indent.addln('');
 }
 
+/// Returns the expression to create an EncodableValue from a host API argument
+/// with the given [variableName] and types.
+String _wrappedHostApiArgumentExpression(Root root, String variableName,
+    TypeDeclaration dartType, HostDatatype hostType) {
+  final String encodableValue;
+  if (!hostType.isBuiltin &&
+      root.classes.any((Class c) => c.name == dartType.baseName)) {
+    final String operator = hostType.isNullable ? '->' : '.';
+    encodableValue =
+        'flutter::EncodableValue($variableName${operator}ToEncodableList())';
+  } else if (!hostType.isBuiltin &&
+      root.enums.any((Enum e) => e.name == dartType.baseName)) {
+    final String nonNullValue =
+        hostType.isNullable ? '(*$variableName)' : variableName;
+    encodableValue = 'flutter::EncodableValue((int)$nonNullValue)';
+  } else {
+    final String operator = hostType.isNullable ? '*' : '';
+    encodableValue = 'flutter::EncodableValue($operator$variableName)';
+  }
+
+  if (hostType.isNullable) {
+    return '$variableName ? $encodableValue : flutter::EncodableValue()';
+  }
+  return encodableValue;
+}
+
+// Writes the code to declare and populate a variable called  [argName] to use
+// as a parameter to an API method call from an existing EncodableValue
+// variable called [encodableArgName] which corresponds to [arg] in the API
+// definition.
+void _writeEncodedArgumentExtraction(
```

Comment:
nit: I'm not sure what "extraction" means based on the docstring.  Is that a term I'm unfamiliar with?

Reply:
Yea, thats a hard one to name.  I think unwrapping is more in line with what we call it elsewhere.  The names not going to be clear, we just need to make sure what you just said is conveyed in the docstring.